src/features/hotspots/root/ShortcutNav.tsx

```diff
@@ -0,0 +1,394 @@
+          case 'explore':
+            return <Globe color={color()} />
+          case 'search':
+            return <Search color={color()} />
+        }
+      }
+      return (
+        <TouchableOpacityBox
+          alignItems="center"
+          hitSlop={{ top: 8, bottom: 8, left: 4, right: 4 }}
+          onPress={handlePress(item)}
+          justifyContent="center"
+          borderRadius="round"
+          width={ITEM_SIZE}
+          height={ITEM_SIZE}
+          marginRight={ITEM_MARGIN}
+          backgroundColor={selected ? 'white' : 'purpleMuted'}
+        >
+          {getIcon()}
+        </TouchableOpacityBox>
+      )
+    },
+    [colors, handlePress, isSelected, selectedItem],
+  )
+
+  type ListItem = { item: Hotspot | GlobalOpt; index: number }
+  const renderItem = useCallback(
+    ({ item, index }: ListItem) => {
+      if (IS_GLOBAL_OPT(item)) {
+        return renderGlobalOpt(item as GlobalOpt)
+      }
+      return renderHotspot(item, index)
+    },
+    [renderGlobalOpt, renderHotspot],
+  )
+
+  const keyExtractor = useCallback((item: GlobalOpt | Hotspot) => {
+    if (IS_GLOBAL_OPT(item)) {
+      return item
+    }
+    return item.address
+  }, [])
+
+  useEffect(() => {
+    const index = data.findIndex((d) => isSelected(d, selectedItem))
+    scroll(index, Platform.OS === 'android')
+  }, [selectedItem, data, scroll, isSelected, scrollOffsets])
+
+  const contentContainerStyle = useMemo(
+    () => ({ paddingHorizontal: wp(50) - ITEM_SIZE / 2 }),
+    [],
+  )
+
+  const handleScroll = useCallback(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+      scrollOffset.current = event.nativeEvent.contentOffset.x
+    },
+    [],
+  )
+
+  const handleScrollMomentum = useCallback(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+      let snapIndex = 0
+      const scrolledOffset = event.nativeEvent.contentOffset.x
+
+      let highIndex = scrollOffsets.findIndex(
+        (offset) => offset > scrolledOffset,
+      )
+      if (highIndex === -1) {
+        // couldn't find an offset larger than scrolled position
+        // snap to last position
+        highIndex = scrollOffsets.length - 1
+      }
+
+      if (highIndex > 0) {
+        const lowIndex = highIndex - 1
+        const lowDiff = scrolledOffset - scrollOffsets[lowIndex]
+        const highDiff = scrollOffsets[highIndex] - scrolledOffset
+        snapIndex = lowDiff < highDiff ? lowIndex : highIndex
+      }
+      const nextItem = data[snapIndex]
+      const nextPos = scrollOffsets[snapIndex]
+
+      if (nextItem === selectedItem) return
+
+      snapPos.current = nextPos
+      onItemSelected(nextItem)
+    },
+    [data, onItemSelected, scrollOffsets, selectedItem],
+  )
+  return (
+    <Box height={SHORTCUT_NAV_HEIGHT} backgroundColor="primaryBackground">
+      <Box top={-43} left={0} right={0} height={43} position="absolute">
+        <Mask width="100%" color={colors.primaryBackground} />
+      </Box>
+      <FlatList
```

Comment:
This works fantastically, good math for all those calculations haha

Reply:
ha, thanks! It was a bit tricky to figure out.